src/main/scala/com/microsoft/azure/iot/iothubreact/checkpointing/Configuration.scala

```diff
@@ -4,21 +4,24 @@ package com.microsoft.azure.iot.iothubreact.checkpointing
 
 import java.util.concurrent.TimeUnit
 
+import com.microsoft.azure.iot.iothubreact.checkpointing.backends.AzureBlob
+import com.microsoft.azure.iot.iothubreact.checkpointing.backends.cassandra.lib.Auth
 import com.typesafe.config.{Config, ConfigFactory}
 
 import scala.concurrent.duration._
 import scala.language.postfixOps
+import scala.util.Try
 
 /** Hold IoT Hub stream checkpointing configuration settings
   */
-private[iothubreact] object Configuration {
+final class Configuration(loader: Config = ConfigFactory.load) {
```

Comment:
since we are introducing DI and mocking, I would use an interface and mock the interface instead. I would also like not to change classes visibility for the sake of testing, to avoid exposing API not meant to be accessible.
I did some refactoring on top of your PR, here's how the interface would look like:

```
trait ICPConfiguration {
  val isEnabled                 : Boolean
  val storageNamespace          : String
  val checkpointBackendType     : String
  val checkpointFrequency       : FiniteDuration
  val checkpointRWTimeout       : FiniteDuration
  val checkpointCountThreshold  : Int
  val checkpointTimeThreshold   : FiniteDuration
  val azureBlobEmulator         : Boolean
  val azureBlobConnectionString : String
  val azureBlobLeaseDuration    : FiniteDuration
  val cassandraCluster          : String
  val cassandraReplicationFactor: Int
  val cassandraAuth             : Option[Auth]
}
```